src/webserver/content_security_policy.rs

```diff
@@ -1,40 +1,101 @@
-use std::fmt::Display;
-
+use actix_web::http::header::{
+    HeaderName, HeaderValue, TryIntoHeaderPair, CONTENT_SECURITY_POLICY,
+};
+use actix_web::HttpResponseBuilder;
 use awc::http::header::InvalidHeaderValue;
 use rand::random;
+use std::fmt::{Display, Formatter};
+
+pub const DEFAULT_CONTENT_SECURITY_POLICY: &str = "script-src 'self' 'nonce-{NONCE}'";
 
-#[derive(Debug, Clone, Copy)]
+#[derive(Debug, Clone)]
 pub struct ContentSecurityPolicy {
     pub nonce: u64,
+    policy: String,
 }
 
-impl Default for ContentSecurityPolicy {
-    fn default() -> Self {
-        Self { nonce: random() }
+impl ContentSecurityPolicy {
+    pub fn new<S: Into<String>>(policy: S) -> Self {
+        Self {
+            nonce: random(),
+            policy: policy.into(),
+        }
```

Comment:
This copies the CSP template string and reparses it on every request.  I was suggesting to parse it just once at the beginning when we parse the configuration, and then reuse the same parsed string without copying it in memory on every request.